### PR TITLE
How to use Form Filling PDF and Functionality use cases to the .NET MAUI PDF Viewer

### PR DESCRIPTION
**Pull request by deepika134**
An example added for How to use Form Filling PDF and Functionality use cases to the .NET MAUI PDF Viewer.

### CODE REVIEW

File: FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+           UIKit.UIApplication.SharedApplication.SendAction(new ObjCRuntime.Selector("resignFirstResponder"), null, null, null);
+#endif
+                }
+            }
+        }
+
+        /// <summary>
+        /// Handle DatePicker's Ok button action.
+        /// </summary>
+        private void datePicker_OkButtonClicked(object sender, EventArgs e)
```
Comment (Deepak1799) [inline FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs:55]: datePicker_OkButtonClicked

Please maintain naming conventions correctly in all the places

File: FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    public partial class MainPage : ContentPage
+    {
+        public MainPage()
+        {
+            InitializeComponent();
+            // Load the PDF document.
+            Stream? stream = typeof(MainPage).GetTypeInfo().Assembly.GetManifestResourceStream("FormFillWithUseCase.Assets.workshop_registration.pdf");
+            pdfViewer.LoadDocument(stream);
+        }
+        /// <summary>
```
Comment (Deepak1799) [inline FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs:17]: Please provide one line between methods

File: FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        /// Share the PDF form externally via platform's share dialog
+        /// </summary>
+        private async void shareButton_Clicked(object sender, EventArgs e)
+        {
+            bool isFormDataValid = await ValidateFormData();
+            if (isFormDataValid)
+            {
+                MemoryStream outputStream = new MemoryStream();
```
Comment (Deepak1799) [inline FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs:90]: Please check whether any resources need to be disposed in the method after use

File: FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+        /// <summary>
+        /// Handle DatePicker's Cancel button action.
+        /// </summary>
+        private void datePicker_CancelButtonClicked(object sender, EventArgs e)
+        {
+            datePickerGrid.IsVisible = false;
+        }
+        /// <summary>
+        /// Share the PDF form externally via platform's share dialog
```
Comment (Deepak1799) [inline FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs:83]: image

There is no platform specific codes are used here

File: FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+        /// </summary>
+        private async void shareButton_Clicked(object sender, EventArgs e)
+        {
+            bool isFormDataValid = await ValidateFormData();
+            if (isFormDataValid)
+            {
+                MemoryStream outputStream = new MemoryStream();
+                await pdfViewer.SaveDocumentAsync(outputStream);
```
Comment (Deepak1799) [inline FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs:91]: Please include comments for important lines of code

File: FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+                    await Share.Default.RequestAsync(shareFileRequest);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Perform validations on the form data filled
+        /// </summary>
+        /// <returns>"True", if all the validations are passed. Otherwise, "False".</returns>
+        async Task<bool> ValidateFormData()
```
Comment (Deepak1799) [inline FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs:111]: Please include comments for important lines of code

File: FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+
+                    dateOfBirthField.Text = date;
+                }
+            }
+            datePickerGrid.IsVisible = false;
+        }
+
+        /// <summary>
+        /// Handle DatePicker's Cancel button action.
+        /// </summary>
```
Comment (Deepak1799) [inline FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs:77]: In summary please cover the parameters also

File: FormFillWithUseCase/FormFillWithUseCase/Properties/launchSettings.json [hunk 8]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (Deepak1799) [inline FormFillWithUseCase/FormFillWithUseCase/Properties/launchSettings.json:1]: As we removed the default template code in the sample, can we remove the unwanted images from the folder?
image

**Review by Deepak1799 — CHANGES_REQUESTED** [review]

**Review by Deepak1799 — CHANGES_REQUESTED** [review]
Please ensure the sample is working in all the 4 platforms supported in MAUI

File: FormFillWithUseCase/FormFillWithUseCase/Properties/launchSettings.json (hunk 8 above)
Comment (deepika134) [reply FormFillWithUseCase/FormFillWithUseCase/Properties/launchSettings.json:1]: I removed the unused image in the image folder.

File: FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs (hunk 7 above)
Comment (deepika134) [reply FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs:77]: Added

File: FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs (hunk 6 above)
Comment (deepika134) [reply FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs:111]: changed.

File: FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs (hunk 5 above)
Comment (deepika134) [reply FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs:91]: added .

File: FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs (hunk 4 above)
Comment (deepika134) [reply FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs:83]: Changed.

File: FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs (hunk 3 above)
Comment (deepika134) [reply FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs:90]: Removed.

File: FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs (hunk 2 above)
Comment (deepika134) [reply FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs:17]: Added the Space.

File: FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs (hunk 1 above)
Comment (deepika134) [reply FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml.cs:55]: Changed the names as per the naming conventions.

**Comment by deepika134** [comment]
> Please ensure the sample is working in all the 4 platforms supported in MAUI

Ensured the sample in 4 Platforms . It's working fine.

File: FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml [hunk 9]
```diff
@@ -0,0 +1,20 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:viewer="clr-namespace:Syncfusion.Maui.PdfViewer;assembly=Syncfusion.Maui.PdfViewer"
+             xmlns:picker="clr-namespace:Syncfusion.Maui.Picker;assembly=Syncfusion.Maui.Picker"
+             xmlns:button="clr-namespace:Syncfusion.Maui.Buttons;assembly=Syncfusion.Maui.Buttons"
+             x:Class="FormFillWithUseCase.MainPage">
+    <Grid>
+        <viewer:SfPdfViewer x:Name="pdfViewer" FormFieldFocusChanged="PdfViewer_FormFieldFocusChanged"/>
```
Comment (Deepak1799) [inline FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml:9]: Please add description comments for code. Example: https://help.syncfusion.com/maui-toolkit/tabview/getting-started#populate-tab-items-in-tab-view

**Review by Deepak1799 — CHANGES_REQUESTED** [review]

File: FormFillWithUseCase/FormFillWithUseCase/FormFillWithUseCase.csproj [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+	<ItemGroup>
+	  <EmbeddedResource Include="Assets\workshop_registration.pdf" />
+	</ItemGroup>
+
+	<ItemGroup>
+		<PackageReference Include="Microsoft.Maui.Controls" Version="$(MauiVersion)" />
+		<PackageReference Include="Microsoft.Maui.Controls.Compatibility" Version="$(MauiVersion)" />
+		<PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="8.0.0" />
+		<PackageReference Include="Syncfusion.Maui.Buttons" Version="27.1.51" />
```
[the remark this answers is not in the document]
Comment (deepika134) [reply FormFillWithUseCase/FormFillWithUseCase/FormFillWithUseCase.csproj:72]: Addressed

File: FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml (hunk 9 above)
Comment (deepika134) [reply FormFillWithUseCase/FormFillWithUseCase/MainPage.xaml:9]: Added

File: FormFillWithUseCase/.vs/FormFillWithUseCase/v17/DocumentLayout.json [hunk 11]
```diff
@@ -0,0 +1,88 @@
+{
```
[the remark this answers is not in the document]
Comment (deepika134) [reply FormFillWithUseCase/.vs/FormFillWithUseCase/v17/DocumentLayout.json]: Removed

**Review by Deepak1799 — APPROVED** [review]

**Pull request merged**